src/utils/userLocation.ts

```diff
@@ -1,52 +1,83 @@
-import firebase, { UserInfo } from "firebase/app";
+import firebase from "firebase/app";
 
 import { AnyRoom } from "types/rooms";
 
 import { updateUserProfile } from "pages/Account/helpers";
 import { useInterval } from "hooks/useInterval";
 
 import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
+import { openRoomUrl } from "./url";
 
 const LOCATION_INCREMENT_SECONDS = 10;
 const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
 
 type LocationData = Record<string, number>;
 
-export const updateLocationData = (
-  userId: string,
-  newLocationData: LocationData = {}
-) => {
+type UpdateLocationDataProps = {
+  userId: string;
+  newLocationData: LocationData;
+};
+
+export const updateLocationData = ({
+  userId,
+  newLocationData,
+}: UpdateLocationDataProps) => {
   updateUserProfile(userId, {
     lastSeenAt: getCurrentTimeInMilliseconds(),
     lastSeenIn: newLocationData,
   });
 };
 
-export interface TrackLocationProps {
+type SetNewLocationDataProps = {
   userId: string;
   locationName: string;
   lastSeenIn?: LocationData;
-}
+};
 
-export const trackLocationEntered = ({
+export const setNewLocationData = ({
   userId,
   locationName,
-}: TrackLocationProps) => {
-  updateLocationData(userId, {
-    [locationName]: getCurrentTimeInMilliseconds(),
+}: SetNewLocationDataProps) => {
+  updateLocationData({
+    userId,
+    newLocationData: {
+      [locationName]: getCurrentTimeInMilliseconds(),
+    },
   });
 };
 
-export const enterVenue = (venueId: string) => openUrl(venueInsideUrl(venueId));
+type UpdateCurrentLocationDataProps = {
+  userId: string;
+  profileLocationData: LocationData;
+};
+
+export const updateCurrentLocationData = ({
+  userId,
+  profileLocationData,
+}: UpdateCurrentLocationDataProps) => {
+  const [locationName] = Object.keys(profileLocationData);
```

Comment:
This is just going to set `locationName` to be the first array item returned from `Object.keys` yeah? That seems, wrong? Maybe?

What's the intended effect here?

Reply:
The intended effect is to update the current location, without rewriting it. profileLocationData can only have 1 key at any point of time

Reply:
Right, that makes sense. Perhaps adding a comment saying something like that would be useful, as it wasn't immediately obvious to me.

Reply:
See also https://github.com/sparkletown/sparkle/pull/1129#discussion_r569931904

Reply:
@mike-lvov can you please add a comment to the code here explaining what the intent/reasoning of this line is, so that future developers understand why we are only updating a single key here, and the assumptions/etc this was written with.